### PR TITLE
[mds-api-server] Remove redundancy using standard authorizer claims

### PR DESCRIPTION
**Pull request by mdurling**
Each API implemented its own authorization header parser. Converted APIs to utilize the standard authorizer claims provided by `mds-api-authorizer` and `mds-api-server and removed of all the various and largely redundant `getAuth` functions.

## Impacts
- [x] Provider
- [x] Agency
- [x] Audit
- [x] Policy
- [x] Compliance

### CODE REVIEW

File: packages/mds-api-authorizer/tests/index.spec.ts [hunk 1]
```diff
@@ -9,13 +9,13 @@ const ADMIN_AUTH = `basic ${Buffer.from(`${PROVIDER_UUID}|${PROVIDER_SCOPES}`).t
 describe('Test API Authorizer', () => {
   it('No Authorizaton', done => {
     const authorizer = AuthorizationHeaderApiAuthorizer({} as express.Request)
-    test.value(authorizer).is({})
+    test.value(authorizer).is(null)
```
Comment (janedotx) [inline packages/mds-api-authorizer/tests/index.spec.ts]: Just curious, what's the motivation for this change?

Comment (mdurling) [reply packages/mds-api-authorizer/tests/index.spec.ts]: Makes the type guarding more predictable because it forces authorizers to return claims and not partial claims and is more reflective of what actually happens in the authorizer.

Comment (janedotx) [reply packages/mds-api-authorizer/tests/index.spec.ts]: 👍
